### PR TITLE
Advanced get info

### PR DESCRIPTION
**Pull request by AlexandraKapp**
add enum values to meta data to ´field.enum_info()´ and ´field.get_info()´
add description to arguments of a field ´field.arguments_info()´ and ´field.get_info()´

### CODE REVIEW

File: datenguidepy/query_builder.py [hunk 1]
```diff
@@ -243,12 +245,29 @@ def _no_none_values(base_function, dict_content, sub_key) -> Optional[str]:
         return base_function(value)
 
     def _arguments_info_helper(self, meta_fields) -> Optional[str]:
```
[the remark this answers is not in the document]
Comment (AlexandraKapp) [reply datenguidepy/query_builder.py]: hm.. I learned to name the recursion function that is being called by the main function "_helper". Do you have a recommendation for sth else?

File: datenguidepy/query_builder.py [hunk 2]
```diff
@@ -243,12 +245,29 @@ def _no_none_values(base_function, dict_content, sub_key) -> Optional[str]:
...
+        UNDERLINE = "\033[4m"
+        NORMAL = "\033[0m"
         args = meta_fields[self.name].get_arguments()
         arg_list = []
+
         for key, value in args.items():
-            temp_arg = key + str(value)
+            args_field = Field(name=key, parent_field=self, return_type=value[3])
+
+            temp_arg = UNDERLINE + key + NORMAL + ": " + str(value[0])
```
[the remark this answers is not in the document]
Comment (AlexandraKapp) [reply datenguidepy/query_builder.py]: also with "temp" I learned that it is used to symbolize, that the value of a variable has no real semantic meaning. In this case: the temp variable keeps changing until the end of the function.

File: datenguidepy/query_builder.py [hunk 3]
```diff
@@ -243,12 +245,29 @@ def _no_none_values(base_function, dict_content, sub_key) -> Optional[str]:
...
+
+            # get type
+            if value[0] == "LIST":
+                temp_arg += " of type " + str(value[2]) + "(" + str(value[3]) + ")"
+            else:
+                temp_arg += "(" + str(value[1]) + ")"
+
+            # get enum values
+            if "ENUM" in value:
+                enum_values = args_field.enum_info()
```
[the remark this answers is not in the document]
Comment (AlexandraKapp) [reply datenguidepy/query_builder.py]: yes, good point with the lifetime. I changed sth later and forgot to adapt that ;)
yea.. I was also thinking about not calling the function, but decided that this was better, so if we change the formatting we only need to change it once..

File: datenguidepy/query_builder.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (KonradUdoHannes) [reply datenguidepy/query_builder.py]: I see the point, but think it is a difficult convention to use effectively. I think it only helps a lot if it is used in that way throughout the code base and it requires the reader of a particular piece of code to be aware of this convention. It also suffers in situations where one function needs two helpers all of the sudden or the helper becomes useful for more than one parent function.
Sometimes an alternative is to define the helper functions inside the scope of the original function which makes their local use very clear but readability may suffer due to an increased indentation level. In any case I prefer if the function name reflects more information about how it helps for instance with the recursion. But in the end this should not be an issue for this commit and could be discussed in general should we feel inclined to do so.

File: datenguidepy/query_builder.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (KonradUdoHannes) [reply datenguidepy/query_builder.py]: I disagree with both points. I don't believe the variable doesn't have semantic meaning as it is used to build string information for individual arguments. Something along the the lines of `single_arg_desc` or similar might be appropriate. The only case where I would agree that there is not semantic meaning to a variable are the for loops that don't use their counter in the typical `for _ in range(n)` way. Here the `_` is the convention indicating that this variable is not used for anything, but the `temp_arg` is used for a clear purpose and the name neglected the opportunity to make the code more readable.
For the second point I would argue that all variables can change all the time, so I don't see why variables that change more often should have less informative names. Misunderstanding their purpose may even be more troublesome as they are used a lot.

**Pull request merged**